### PR TITLE
add integration tests

### LINKED ISSUE #214: Add Merlin example notebooks as integration tests with real data

### PR DESCRIPTION
**Pull request by radekosmulski**
Should be good to go now, I left a bunch of comments in the associated issue

resolves #214

### CODE REVIEW

**Comment by radekosmulski** [comment]
Finally! 🥳

This is now passing on ngc, works okay with a 32GB GPU. Both the unit and integration tests run okay.

@bschifferer this is good to go AFAICT 🙂 

```
From https://github.com/NVIDIA-Merlin/Merlin
 * [new ref]         refs/pull/310/head -> add_integration_test
Switched to branch 'add_integration_test'
============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-7.1.1, pluggy-1.0.0
rootdir: /Merlin
plugins: anyio-3.6.1
collected 2 items

tests/examples/unit/test_building_deploying_multi_stage_RecSys.py ..     [100%]

======================== 2 passed in 163.13s (0:02:43) =========================
============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-7.1.1, pluggy-1.0.0
rootdir: /Merlin
plugins: anyio-3.6.1
collected 2 items

tests/examples/integration/test_building_deploying_multi_stage_RecSys.py . [ 50%]
.                                                                        [100%]

======================== 2 passed in 2963.12s (0:49:23) ========================
```

**Review by bschifferer — CHANGES_REQUESTED** [review]
Should we keep it as one unittest? because they depend on each other?

File: tests/examples/unit/test_building_deploying_multi_stage_RecSys.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    assert os.path.isfile("/tmp/examples/item_embeddings.parquet")
+    assert os.path.isfile("/tmp/examples/feature_repo/user_features.py")
+    assert os.path.isfile("/tmp/examples/feature_repo/item_features.py")
+
+@testbook(
+    REPO_ROOT
+    / "examples/Building-and-deploying-multi-stage-RecSys/02-Deploying-multi-stage-RecSys-with-Merlin-Systems.ipynb",
+    execute=False
+)
+def test_nb2(tb):
```
[the remark this answers is not in the document]
Comment (radekosmulski) [reply tests/examples/unit/test_building_deploying_multi_stage_RecSys.py:36]: That's a good point! I didn't think of the dependency, you are right. Will revert to how it was before.

The problem I encountered which prompted this change was that now with running on much bigger data I was running out of GPU RAM when this was a single test. But maybe this was due to some other issue in the code at that time.

Will switch to this being a single test, rerun the code on a 32GB GPU on ngc and if there will be issues will see if I can shutdown the first NB or clear the state via some other means when running the 2nd one.

**Comment by radekosmulski** [comment]
Finally got this to pass on an ngc instance with a 32 GPU 🙂 

The integration test is formatted rather weirdly, but that is because regardless of what I tried, I was getting an OOM error. If the first notebook is used in the decorator, no matter what I tried, the state was retained inside the function after the execution of `tb1` finishes. The only way I managed to free up GPU memory was through the use of the context manager.

**Comment by bschifferer** [comment]
rerun tests

**Review by rnyak — APPROVED** [review]

**Review by bschifferer — APPROVED** [review]

**Comment by bschifferer** [comment]
rerun tests

**Comment by bschifferer** [comment]
rerun tests

**Comment by bschifferer** [comment]
rerun tests

**Comment by bschifferer** [comment]
rerun tests

**Comment by bschifferer** [comment]
rerun tests

**Comment by bschifferer** [comment]
rerun tests

File: examples/Building-and-deploying-multi-stage-RecSys/01-Building-Recommender-Systems-with-Merlin.ipynb [hunk 2]
```diff
@@ -126,7 +126,24 @@
...
+      "application/javascript": [
```
Comment (jperez999) [inline examples/Building-and-deploying-multi-stage-RecSys/01-Building-Recommender-Systems-with-Merlin.ipynb:129]: remove this change and the tests seem to work. I would replace this code box with a text box that just tells them the command to run.

**Review by jperez999 — CHANGES_REQUESTED** [review]

**Pull request merged**